controllers/controllers_suite_test.go

```diff
@@ -371,3 +433,137 @@ func setupMachineCRDs() {
+}
+
+// labelMachineFailuredomain add cloudstackfailuredomain info in the labels.
+func labelMachineFailuredomain(CSMachine *infrav1.CloudStackMachine, CSFailureDomain1 *infrav1.CloudStackFailureDomain) {
+	key := client.ObjectKey{Namespace: dummies.CSCluster.Namespace, Name: CSMachine.Name}
+	Eventually(func() error {
+		return k8sClient.Get(ctx, key, CSMachine)
+	}, timeout).Should(BeNil())
+
+	// Set owner ref from CAPI machine to CS machine and patch back the CS machine.
+	Eventually(func() error {
+		ph, err := patch.NewHelper(CSMachine, k8sClient)
+		Ω(err).ShouldNot(HaveOccurred())
+		CSMachine.Labels["cloudstackfailuredomain.infrastructure.cluster.x-k8s.io/name"] = CSFailureDomain1.Name
+		return ph.Patch(ctx, CSMachine, patch.WithStatusObservedGeneration{})
+	}, timeout).Should(Succeed())
+}
+
+// labelMachineDeploymentName add machine deployment info in the labels.
+func labelMachineDeploymentName(CSMachine *infrav1.CloudStackMachine, machineDeploymentName string) {
+	key := client.ObjectKey{Namespace: dummies.CSCluster.Namespace, Name: CSMachine.Name}
+	Eventually(func() error {
+		return k8sClient.Get(ctx, key, CSMachine)
+	}, timeout).Should(BeNil())
+
+	// Set owner ref from CAPI machine to CS machine and patch back the CS machine.
+	Eventually(func() error {
+		ph, err := patch.NewHelper(CSMachine, k8sClient)
+		Ω(err).ShouldNot(HaveOccurred())
+		CSMachine.Labels["cluster.x-k8s.io/deployment-name"] = machineDeploymentName
+		return ph.Patch(ctx, CSMachine, patch.WithStatusObservedGeneration{})
+	}, timeout).Should(Succeed())
+}
+
+func setCSMachineTemplateCRDs(CSMachineTemplate *infrav1.CloudStackMachineTemplate) {
+	//  Create cloudstack machine template.
+	Ω(k8sClient.Create(ctx, CSMachineTemplate)).Should(Succeed())
+
+	// Fetch the CAPI Machine Deployment that was created.
+	key := client.ObjectKey{Namespace: dummies.ClusterNameSpace, Name: CSMachineTemplate.Name}
+	Eventually(func() error {
+		return k8sClient.Get(ctx, key, CSMachineTemplate)
+	}, timeout).Should(BeNil())
+}
+
+func setMachineOwnerReference(CSMachine *infrav1.CloudStackMachine, ownerRef metav1.OwnerReference) {
+	key := client.ObjectKey{Namespace: dummies.CSCluster.Namespace, Name: CSMachine.Name}
+	Eventually(func() error {
+		return k8sClient.Get(ctx, key, CSMachine)
+	}, timeout).Should(BeNil())
+
+	// Set owner ref from CAPI machine to CS machine and patch back the CS machine.
+	Eventually(func() error {
+		ph, err := patch.NewHelper(CSMachine, k8sClient)
+		Ω(err).ShouldNot(HaveOccurred())
+		CSMachine.OwnerReferences = append(CSMachine.OwnerReferences, ownerRef)
+		return ph.Patch(ctx, CSMachine, patch.WithStatusObservedGeneration{})
+	}, timeout).Should(Succeed())
+}
+
+func setMachineAnnotation(CSMachine *infrav1.CloudStackMachine, annotationKey string, annotationValue string) {
+	key := client.ObjectKey{Namespace: dummies.CSCluster.Namespace, Name: CSMachine.Name}
+	Eventually(func() error {
+		return k8sClient.Get(ctx, key, CSMachine)
```

Comment:
Updating the Machine object passed in by ref seems like an unexpected (to the caller) side-effect of calling this func.

Reply:
No, bad comment on my part.  Agreed that the function name should have made it clear that the CSMachine would be updated.